### PR TITLE
New test for checking configs persist.

### PR DESCRIPTION
**Pull request by petedyerarm**
Added new test to check that the config changes persist across a reboot.
This tests the hostname changes and the wifi setup.

Fixed error in test-enable-wifi.py which meant that WiFi wasn't always
enabled.

### CODE REVIEW

File: ci/lava/tests/test-config-across-reboot.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        test_command = 'sh -l -c "connmanctl disable {}"'.format(technology)
+
+        err, stdout, stderr = execute_helper.send_mbl_cli_command(
+            ["shell", test_command], TestConfigAcrossReboot.dut_address
+        )
+
+        # Wait for the interface to be removed and the routing tables
+        # etc to be updated
+        time.sleep(30)
+        self._ifconfig(execute_helper)
```
Comment (jjohnson-arm) [inline ci/lava/tests/test-config-across-reboot.py]: What is this for?

File: ci/lava/tests/test-config-across-reboot.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        err, hostname = self._get_hostname(dut, dut_addr, execute_helper)
+
+        assert err == 0
+        assert hostname is not None
+        assert TestConfigAcrossReboot.newHostname is not None
+        assert hostname == TestConfigAcrossReboot.newHostname
+
+    def test_get_wlan_ip_after_reboot(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
+        self._ifconfig(execute_helper)
```
Comment (jjohnson-arm) [inline ci/lava/tests/test-config-across-reboot.py]: Why this?

File: ci/lava/tests/test-config-across-reboot.py (hunk 1 above)
Comment (petedyerarm) [reply ci/lava/tests/test-config-across-reboot.py]: This is debug code that will show the ifconfig of the DUT in the Lava Log - might be useful to understand the reason if the IP address part of the test fails.

File: ci/lava/tests/test-config-across-reboot.py (hunk 2 above)
Comment (petedyerarm) [reply ci/lava/tests/test-config-across-reboot.py]: This is debug code that will show the ifconfig of the DUT in the Lava Log - might be useful to understand the reason if the IP address part of the test fails.

File: ci/lava/tests/test-config-across-reboot.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                    hostname = line
+        if hostname:
+            print("_get_hostname: " + hostname)
+        else:
+            print("_get_hostname: No hostname found")
+
+        return err, hostname
+
+    def _get_wlan_ip(self, execute_helper):
+        """Perform the test on the DUT via the mbl-cli."""
```
Comment (jjohnson-arm) [inline ci/lava/tests/test-config-across-reboot.py]: Update comment
```suggestion
        """Retrieve WLAN IP address from the DUT via the mbl-cli."""
```

**Pull request merged**